AiTableTopGameMaster.ConsoleApp/Clients/ConsoleChatClient.cs

```diff
@@ -48,16 +50,53 @@ public async Task<ChatHistory> ChatAsync(ChatHistory history, CancellationToken
         
         try
         {
-            // Use the agent framework properly for chatting
-            await foreach (AgentResponseItem<ChatMessageContent> response in agent.InvokeAsync(history, cancellationToken: cancellationToken))
+            const int maxRevisionAttempts = 3;
+            int revisionAttempt = 0;
+            
+            while (revisionAttempt < maxRevisionAttempts)
             {
-                ChatMessageContent message = response.Message;
-                history.Add(message);
+                // Generate response from the game master agent
+                string gameMasterResponse = await GenerateGameMasterResponseAsync(history, cancellationToken);
                 
-                log.LogInformation("{Agent}: {Content}", agent.Name, message.Content);
+                // Review the response
+                OutputReviewResult reviewResult = await outputReviewer.ReviewOutputAsync(gameMasterResponse, cancellationToken);
                 
-                console.Markup($"{DisplayHelpers.AI}{agent.Name}:[/] ");
-                console.MarkupLineInterpolated($"{message.Content}");
+                if (reviewResult.IsAcceptable)
```

Comment:
We could probably remove some code if we make this an or statement that says that we'll go with it if we've exceeded our maximum count.